Feature/add env var secrets for production

mattwendzina opened the pull request:
Closes #

## 📝 Description

Add the secrets for running Algolia on production and a few other refactors to tidy up Algolia env vars across the codebase.

## ⛳️ Current behavior (updates)

There aren't the correct secrets set for production and Algolia can't currently be used in production.

## 🚀 New behavior

- Adds env vars in the docker files and deployment ymls.
- Updates env vars in other workflows (chromatic, vitest, playwright)
- Updates env vars in env.sample and README.md
- Renames the env vars to be a bit more accurate in terms of their description
- Removes non-nullable operators due to TS linting error
- Throws a 404 and logs an error for routes where Algolia is used if env-vars are undefined
- Removes storybook and vite env vars and updates these with the standard ones

## 💣 Is this a breaking change (Yes/No):

Env vars will need updating

## 📝 Additional Information

Full list of Algolia env vars:
```
ALGOLIA_APP_ID
ALGOLIA_API_KEY_SEARCH
ALGOLIA_API_KEY_ADMIN
ALGOLIA_PRIMARY_ALL_RECORDS_INDEX - replaces ALGOLIA_PRIMARY_PRODUCTS_INDEX
ALGOLIA_PRIMARY_COLLECTIONS_INDEX
ALGOLIA_ALL_RECORDS_INDEX_GROUPED_BY_ID - replaces ALGOLIA_PRODUCTS_INDEX_GROUPED_BY_ID
ALGOLIA_ALL_RECORDS_INDEX_QUERY_SUGGESTIONS - replaces ALGOLIA_PRIMARY_INDEX_QUERY_SUGGESTIONS
ALGOLIA_HITS_PER_PAGE
```

stevenbister reviewed (changes requested):
Looks good, a couple of eslint errors. I've added the tag to build the app, but I think you'll need to add the secrets into GitHub for it to build properly
https://github.com/Open-Study-College/osc/settings/secrets/actions

mattwendzina commented on packages/osc-ecommerce/app/routes/collections/$slug.tsx:
```diff
@@ -92,19 +93,28 @@ export const loader = async ({ request, params, context }: LoaderArgs) => {
+    } as algoliaEnvVars;
+
+    // If any envVars are undefined throw error
+    Object.keys(algoliaEnvVars).some((envVar) => {
```
Updated to a for of loop

mattwendzina commented:
> Looks good, a couple of eslint errors. I've added the tag to build the app, but I think you'll need to add the secrets into GitHub for it to build properly https://github.com/Open-Study-College/osc/settings/secrets/actions

Updated/add all the right secrets 👍
